src/PotentialTypes/BasisPotentials/potential_energy.jl

```diff
@@ -0,0 +1,7 @@
+function potential_energy(A::AbstractSystem, p::BasisPotential)
+    d = evaluate_basis(A, p.basis_params)
+    e = 0.0
+    for di in d
```

Comment:
If you want, this whole function could be
```julia
function potential_energy(A::AbstractSystem, p::BasisPotential)
    sum(dot(di, p.coefficients) for di in evaluate_basis(A, p.basis_params))
end
```
Note that the current implementation doesn't actually return anything.

Reply:
Brilliant.